packages/grid/x-data-grid/src/components/cell/GridActionsCell.tsx

```diff
@@ -16,76 +17,198 @@ interface TouchRippleActions {
   stop: (event: any, callback?: () => void) => void;
 }
 
-type GridActionsCellProps = Pick<GridRenderCellParams, 'colDef' | 'id' | 'api' | 'hasFocus'> &
-  Pick<GridMenuProps, 'position'>;
+interface GridActionsCellProps extends Omit<GridRenderCellParams, 'value' | 'formattedValue'> {
+  value?: GridRenderCellParams['value'];
+  formattedValue?: GridRenderCellParams['formattedValue'];
+  position?: GridMenuProps['position'];
+}
 
 const GridActionsCell = (props: GridActionsCellProps) => {
+  const {
+    colDef,
+    id,
+    api,
+    hasFocus,
+    isEditable,
+    field,
+    value,
+    formattedValue,
+    row,
+    rowNode,
+    cellMode,
+    getValue,
+    tabIndex,
+    position = 'bottom-end',
+    focusElementRef,
+    ...other
+  } = props;
+  const [focusedButtonIndex, setFocusedButtonIndex] = React.useState(-1);
   const [open, setOpen] = React.useState(false);
+  const apiRef = useGridApiContext();
+  const rootRef = React.useRef<HTMLDivElement>(null);
   const buttonRef = React.useRef<HTMLButtonElement>(null);
-  const touchRippleRefs = React.useRef<Record<number, TouchRippleActions | null>>({});
+  const ignoreCallToFocus = React.useRef(false);
+  const touchRippleRefs = React.useRef<Record<string, TouchRippleActions | null>>({});
   const menuId = useId();
   const buttonId = useId();
   const rootProps = useGridRootProps();
-  const { colDef, id, api, hasFocus, position = 'bottom-end' } = props; // TODO apply the rest to the root element
 
   React.useLayoutEffect(() => {
     if (!hasFocus) {
       Object.entries(touchRippleRefs.current).forEach(([index, ref]) => {
         ref?.stop({}, () => {
-          delete touchRippleRefs.current[index as any];
+          delete touchRippleRefs.current[index];
         });
       });
     }
   }, [hasFocus]);
 
+  React.useEffect(() => {
+    if (focusedButtonIndex >= 0) {
+      const child = rootRef.current?.children[focusedButtonIndex] as HTMLElement;
+      child.focus();
+    }
+  }, [focusedButtonIndex]);
```

Comment:
Is it necessary or the following line does the job?
```jsx
tabIndex={focusedButtonIndex === iconButtons.length ? tabIndex : -1}
```

Reply:
`tabIndex` only adds or removes the element from the tab sequence. It's `element.focus` that will call focus.